addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/api/EvohomeApiClient.java

```diff
@@ -40,6 +40,8 @@
 public class EvohomeApiClient {
 
     private static final String APPLICATION_ID = "b013aa26-9724-4dbd-8897-048b9aada249";
+    private static final String CLIENT_ID = "4a231089-d2b6-41bd-a5eb-16a0a422b999";
```

Comment:
Hm, client secrets and client ids do not belong into open source code. See also this discussion: https://www.eclipse.org/forums/index.php?t=msg&th=1096718.
I'm not familiar with how evohome works, but ideally, every user should request his own client id/secret and provide it as a configuration parameter.

Reply:
Yeah, that would be great. Unfortunately, as for the last communication I had with Honeywell, they are not allowing consumers to create these for evohome. It is still following the reference implementation of the python client (https://github.com/watchforstock/evohome-client/blob/master/evohomeclient2/__init__.py). We could do this by means of binding configuration but it feels like kind of like annoying the end user for nothing?

For the moment at least it means that the binding is not usable in the current state :-(

A compromise to the situation is to fill in the current values as a default in the binding config, or to publish the current values on the forum. That way it is still possible to update the values without rebuilding the binary.

Reply:
Well, it is my reference. None of this is officially supported by Honeywell as mentioned before. Though, once this binding is in, I could contact them again, stressing the need for user generated keys.

The python implementation, which was reverse engineer by the author from the evohome app, uses the string `NGEyMzEwODktZDJiNi00MWJkLWE1ZWItMTZhMGE0MjJiOTk5OjFhMTVjZGI4LTQyZGUtNDA3Yi1hZGQwLTA1OWY5MmM1MzBjYg==` which is nothing more than the base64 version of the contents of this commit.

I did however manage to generate a set of keys on the Honeywell site just now, which wasn't possible a while ago. They are aimed for use with Lyric (another Honeywell product), but they might work with evohome. I'll let you know if that works later tonight.

Reply:
Indeed. I reached out to the dev who provided the key pairs for the python binding on info on how to create some key for my/our self. If/when he gets back to me I'll make changes accordingly. Thanks for merging this one until that time!